test/system/255-auto-update.bats

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env bats   -*- bats -*-
+#
+# Tests for automatically update images for containerized services
+#
+
+load helpers
+
+UNIT_DIR="/usr/local/lib/systemd/system"
+
+function setup() {
+    skip_if_remote "systemd tests are meaningless over remote"
+    skip_if_rootless
+
+    basic_setup
+}
+
+function generate_service() {
+    fake_img=$1
+    target_img=$2
+    cmd=$3
+    label=$4
+
+    cname=$(random_string)
+    run_podman tag $fake_img $target_img
+    if [ "$4" ]; then
+        run_podman run -d --name $cname --label "io.containers.autoupdate=$label" $target_img $cmd
+    else
+        run_podman run -d --name $cname $target_img $cmd
+    fi
+
+    run_podman generate systemd --new $cname
+    echo "$output" > "$UNIT_DIR/container-$cname.service"
+    run_podman rm -f $cname
+
+    systemctl daemon-reload
+
+    run systemctl start container-$cname
+    if [ $status -ne 0 ]; then
+        die "Error starting systemd unit $SERVICE_NAME, output: $output"
+    fi
+
+    run systemctl status container-$cname
+    if [ $status -ne 0 ]; then
+        die "Non-zero status of systemd unit $SERVICE_NAME, output: $output"
+    fi
+}
+
+@test "podman auto-update - label io.containers.autoupdate=image" {
+    run generate_service $IMAGE quay.io/libpod/alpine:latest top image
+    cname=${lines[0]}
+    echo $output
+    run_podman inspect --format "{{.Image}}" $cname
+    ori_image=$output
+
+    sleep 2
```

Comment:
I second @ParkerVR's concern. If possible, please use `wait_for_ready` instead.

Reply:
There are two types of sleep in this test. One is wait for the service fully up and another is wait for the container back after we update the image. In my test env, sleep 2 makes it stable and not failed because we check some date too soon. This is also used in the 250-systemd.bats tests. And I am afraid that we can not use wait_for_ready function here as that function only designed to check the container output but not suitable for this situation.

Reply:
This is another good habit to develop: every time you see `sleep`, you think: "this is something that will flake".

Please try this approach instead:
```bash
function _confirm_update() {
    local cname=$1
    local old_iid=$2

    # Image has already been pulled, so this shouldn't take too long
    local timeout=5
    while [[ $timeout -gt 0 ]]; do
        run_podman '?' inspect --format "{{.Image}}" $cname
        if [[ $status != 0 ]]; then
            if [[ $output =~ "no such object:" ]]; then
                # this is ok, it just means the container is being restarted
                :
            fi
        elif [[ $output != $old_iid ]]; then
            return
        fi
        sleep 1
    done

    die "Timed out waiting for $cname to update; old IID=$old_iid"
}
```